system-addon/content-src/components/TopSites/TopSiteForm.jsx

```diff
@@ -1,24 +1,50 @@
 const React = require("react");
 const {actionCreators: ac, actionTypes: at} = require("common/Actions.jsm");
 const {FormattedMessage} = require("react-intl");
+const {TopSiteLink} = require("./TopSite");
 
 const {TOP_SITES_SOURCE} = require("./TopSitesConstants");
 
 class TopSiteForm extends React.PureComponent {
   constructor(props) {
     super(props);
+    const {Topsite} = props;
     this.state = {
-      label: props.label || "",
-      url: props.url || "",
-      validationError: false
+      label: Topsite ? (Topsite.label || Topsite.hostname) : "",
+      url: Topsite ? Topsite.url : "",
+      customScreenshotUrl: Topsite ? Topsite.customScreenshotURL : "",
+      screenshotUrlValidationError: false,
+      validationError: false,
+      pendingScreenshotUpdate: false,
+      // Keep the preview screenshot data in the state to avoid showing
+      // cached results from previous requests.
+      screenshotPreview: null,
+      showCustomScreenshotForm: false,
+      screenshotRequestFailed: false
     };
     this.onLabelChange = this.onLabelChange.bind(this);
     this.onUrlChange = this.onUrlChange.bind(this);
     this.onCancelButtonClick = this.onCancelButtonClick.bind(this);
+    this.onCustomScreenshotUrlChange = this.onCustomScreenshotUrlChange.bind(this);
+    this.onEnableScreenshotUrlForm = this.onEnableScreenshotUrlForm.bind(this);
+    this.onHideCustomScreenshotForm = this.onHideCustomScreenshotForm.bind(this);
```

Comment:
With the 3 of these label/inputs, it would seem natural to move to a component

Reply:
The 3 here I was referring to the 3 label/inputs: title label/input, url label/input, image label/input

Reply:
Created TopSiteEditInputs.jsx
* includes the url, title and custom url inputs
* handles the logic to display the error labels and to focus the inputs
* toggle to show hide the custom screenshot input 
* All the validation and the input change handler are still in the TopsiteForm.